### PR TITLE
chore: improve `from_parts`

### PR DESCRIPTION
**Pull request by yjhmelody**
To reduce unnnessary clone.

### CODE REVIEW

**Comment by SchmErik** [comment]
@risc0/bonsai Could you review this?

File: bonsai/ethereum-relay/src/sdk/client.rs [hunk 1]
```diff
@@ -97,10 +97,10 @@ impl Client {
     }
 
     /// Construct a [Client] from url + api key strings
-    pub fn from_parts(api_url: String, api_key: String) -> Result<Self, ClientError> {
+    pub fn from_parts(api_url: String, api_key: &str) -> Result<Self, ClientError> {
```
Comment (rlukata) [inline bonsai/ethereum-relay/src/sdk/client.rs:100]: did you intend to also change `api_url` to `&str`?

Comment (yjhmelody) [reply bonsai/ethereum-relay/src/sdk/client.rs:100]: Yes, it's not consumed but only need to be borrowed.

File: bonsai/sdk/src/alpha.rs [hunk 2]
```diff
@@ -348,14 +348,14 @@ impl Client {
...
-    /// let api_key = "my_secret_key".to_string();
+    /// let url = "http://api.bonsai.xyz";
+    /// let api_key = "my_secret_key";
     /// bonsai_sdk::Client::from_parts(url, api_key, risc0_zkvm::VERSION)
     ///     .expect("Failed to construct sdk client");
     /// ```
-    pub fn from_parts(url: String, key: String, risc0_version: &str) -> Result<Self, SdkErr> {
-        let client = construct_req_client(&key, risc0_version)?;
-        let url = url.strip_suffix('/').unwrap_or(&url).to_string();
+    pub fn from_parts(url: &str, key: &str, risc0_version: &str) -> Result<Self, SdkErr> {
```
[the remark this answers is not in the document]
Comment (yjhmelody) [reply bonsai/sdk/src/alpha.rs:356]: I set the parameter type based on ownership. Asynchronous and synchronous should not be completely consistent, which has an impact on overhead.

File: bonsai/ethereum-relay/src/sdk/client.rs (hunk 1 above)
Comment (rlukata) [reply bonsai/ethereum-relay/src/sdk/client.rs:100]: I meant that in this `from_parts` function the parameter `api_url` is a `String`. To stay consistent across the library I would prefer if this also was a `&str`.

File: bonsai/sdk/src/alpha.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (mothran) [reply bonsai/sdk/src/alpha.rs:356]: @yjhmelody any thoughts on the feedback above?

[the remark this answers is not in the document]
Comment (yjhmelody) [reply bonsai/sdk/src/alpha.rs:356]: Not yet, I tend to close this PR. I found that the current design of bonsai-sdk is still very early.

**Pull request closed without merging**